pkg/s3/s3driver.go

```diff
@@ -0,0 +1,115 @@
+package s3
+
+import (
+	"fmt"
+
+	// "github.com/argoproj/pkg/file"
+	// s3client "github.com/argoproj/pkg/s3"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+// S3Driver is a driver for Object Storage (MinIO, S3, etc)
+type S3Driver struct {
+	Endpoint              string
+	Region                string
+	Secure                bool
+	AccessKey             string
+	SecretKey             string
+	RoleARN               string
+	UseSDKCreds           bool
+	KmsKeyId              string
+	KmsEncryptionContext  string
+	EnableEncryption      bool
+	ServerSideCustomerKey string
+}
+
+func (s3 *S3Driver) AddFlags(fs *pflag.FlagSet) {
+	fs.StringVar(
+		&s3.Endpoint,
+		"s3-endpoint",
+		"",
+		"swoop s3 endpoint (required; SWOOP_S3_ENDPOINT)",
+	)
+	cobra.MarkFlagRequired(fs, "s3-endpoint")
+	fs.StringVar(
+		&s3.Region,
+		"s3-region",
+		"",
+		"swoop s3 region (SWOOP_S3_REGION)",
+	)
+	fs.BoolVar(
+		&s3.Secure,
+		"s3-secure",
+		false,
+		"swoop s3 secure (SWOOP_S3_SECURE)",
+	)
+	fs.StringVar(
+		&s3.AccessKey,
+		"s3-access-key",
+		"",
+		"swoop s3 access key (required; SWOOP_S3_ACCESS_KEY)",
+	)
+	cobra.MarkFlagRequired(fs, "s3-access-key")
+	fs.StringVar(
+		&s3.SecretKey,
+		"s3-secret-key",
+		"",
+		"swoop s3 secret key (required; SWOOP_S3_SECRET_KEY)",
+	)
+	cobra.MarkFlagRequired(fs, "s3-role-arn")
```

Comment:
I am uncertain if these should be required. Is it possible to get S3 access without them, like when running in AWS with a role that provides it?

Reply:
I think it makes sense to remove the required flag condition here.